src/main/java/com/google/ehub/data/EntertainmentItemDatastore.java

```diff
@@ -0,0 +1,104 @@
+package com.google.ehub.data;
+
+import com.google.appengine.api.datastore.DatastoreService;
+import com.google.appengine.api.datastore.DatastoreServiceFactory;
+import com.google.appengine.api.datastore.Entity;
+import com.google.appengine.api.datastore.PreparedQuery;
+import com.google.appengine.api.datastore.Query;
+import com.google.appengine.api.datastore.Query.CompositeFilterOperator;
+import com.google.appengine.api.datastore.Query.FilterOperator;
+import com.google.appengine.api.datastore.Query.FilterPredicate;
+import com.google.appengine.api.datastore.Query.SortDirection;
+import com.google.ehub.utility.NextIncreasingLexicographicStringUtility;
+import java.util.ArrayList;
+import java.util.List;
+
+
+/**
+ * Manages EntertainmentItems stored in Datastore.
+ */
+public final class EntertainmentItemDatastore {
+  public static final String ENTERTAINMENT_ITEM_KIND = "entertainmentItem";
+  public static final String TITLE_PROPERTY_KEY = "title";
+  public static final String LOWERCASE_TITLE_PROPERTY_KEY = "lowercaseTitle";
+  public static final String DESCRIPTION_PROPERTY_KEY = "description";
+  public static final String IMAGE_URL_PROPERTY_KEY = "imageURL";
+
+  private static DatastoreService datastoreService = DatastoreServiceFactory.getDatastoreService();
+
```

Comment:
Generally static methods are ok for simple use cases (e.g. incrementByOne()) For more complicated classes, a bunch of static methods is not preferable for a couple reasons:

- Your DatastoreService is getting initialized even if you never actually use it
- After someone has used your EntertainmentItemDatastore and is done with it, it will never get garbage collected
- Static can't be faked or mocked

I think at the very least, this should be a non-static class. 

This next part is counter to my advice above, but I think valid in this case: Since there should really only be one instance of your EIDataStore, you can make this a singleton:
```
public class EntertainmentItemDataStore {
  private static EntertainmentItemDataStore instance;

  private EntertainmentItemDataStore() {}

  public static EntertainmentItemDataStore getInstance() {
    if (instance == null) {
     instance = new EntertainmentItemDataStore();
    }
    return instance;
  }
}
```

This prevents too many objects being created since conceptually there should only be one EIDS, not many.

Reply:
I decided to use the Singleton pattern for the class in the last commits.